modules/json/format-json.c

```diff
@@ -414,17 +438,22 @@ tf_flat_json_call(LogTemplateFunction *self, gpointer s,
   TFJsonState *state = (TFJsonState *)s;
   gsize orig_size = result->len;
 
+  ScratchBuffersMarker m;
+  scratch_buffers_mark(&m);
+
   *type = LM_VT_JSON;
   for (gint i = 0; i < args->num_messages; i++)
     {
       gboolean r = tf_flat_json_append(state, result, args->messages[i], args->options);
       if (!r && (args->options->opts->on_error & ON_ERROR_DROP_MESSAGE))
         {
           g_string_set_size(result, orig_size);
-          return;
+          goto exit;
```

Comment:
this changes behaviour, although supporting multiple messages in a $(format-json) is probably a bad idea.

This loop is iterating over the message context, formatting multiple JSON documents if there are multiple messages. This was probably a bad idea which we might be able to get rid off. But I would not change the behaviour with the 'goto exit;' here in neither locations.

Reply:
Sorry, I don't understand your comment.
What did I change with the `goto exit`?
I used it only for reclaiming scratch buffers, which I think is correct even for multiple messages. Everything else should be the same.